tests/test_pagure.py

```diff
@@ -0,0 +1,749 @@
+                               body=body,
+                               status=200,
+                               )
+
+        from_date = datetime.datetime(2020, 3, 8)
+        pagure = Pagure(repository='Project-example')
+        issues = [issues for issues in pagure.fetch(from_date=from_date)]
+
+        self.assertEqual(len(issues), 1)
+        issue = issues[0]
+        self.assertEqual(issue['origin'], 'https://pagure.io/Project-example')
+        self.assertEqual(issue['uuid'], '7dd3642664c8a7e475814b9037277df775657850')
+        self.assertEqual(issue['updated_on'], 1583558174.0)
+        self.assertEqual(issue['category'], CATEGORY_ISSUE)
+        self.assertEqual(issue['tag'], 'https://pagure.io/Project-example')
+        self.assertEqual(issue['data']['assignee']['name'], 'animeshk0806')
+        self.assertEqual(len(issue['data']['comments']), 2)
+        self.assertEqual(issue['data']['comments'][0]['user']['name'], 'animeshk08')
+
+    @httpretty.activate
+    def test_fetch_issues_namespace(self):
+        """Test issues fetch from a repository within a namespace"""
+
+        issue_1 = read_file('data/pagure/pagure_namespace_issue_2')
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_NAMESPACE_ISSUES_URL,
+                               body=issue_1, status=200,
+                               )
+
+        pagure = Pagure(namespace='Test-group', repository='Project-namespace-example')
+        issues = [issues for issues in pagure.fetch()]
+
+        self.assertEqual(len(issues), 2)
+
+        issue = issues[0]
+        self.assertEqual(issue['origin'], 'https://pagure.io/Test-group/Project-namespace-example')
+        self.assertEqual(issue['uuid'], 'bdf90e94bf3b17ed2f75f5e5187e21a62512ca5a')
+        self.assertEqual(issue['updated_on'], 1583509042.0)
+        self.assertEqual(issue['category'], CATEGORY_ISSUE)
+        self.assertEqual(issue['tag'], 'https://pagure.io/Test-group/Project-namespace-example')
+        self.assertEqual(issue['data']['assignee']['name'], 'animeshk08')
+        self.assertEqual(len(issue['data']['comments']), 1)
+        self.assertEqual(issue['data']['comments'][0]['user']['name'], 'animeshk08')
+
+        issue = issues[1]
+        self.assertEqual(issue['origin'], 'https://pagure.io/Test-group/Project-namespace-example')
+        self.assertEqual(issue['uuid'], 'eec4d7bf5c3ca405e39f39a8c6faf616fd4fa425')
+        self.assertEqual(issue['updated_on'], 1583562831.0)
+        self.assertEqual(issue['category'], CATEGORY_ISSUE)
+        self.assertEqual(issue['tag'], 'https://pagure.io/Test-group/Project-namespace-example')
+        self.assertEqual(issue['data']['assignee']['name'], 'animeshk0806')
+        self.assertEqual(len(issue['data']['comments']), 2)
+        self.assertEqual(issue['data']['comments'][1]['user']['name'], 'animeshk08')
+
+    @httpretty.activate
+    def test_fetch_issues_namespace_reaction(self):
```

Comment:
Probably this test isn't needed since the reactions aren't fetched from a specific endpoint.